frb_rust/src/rust2dart.rs

```diff
@@ -139,3 +145,120 @@ impl<T: IntoDart> StreamSink<T> {
         self.rust2dart().close_stream()
     }
 }
+
+/// Basically the Into trait.
+/// We need this separate trait because we need to implement int for Vec<T> etc.
+pub trait IntoIntoDart<D> {
+    fn into_into_dart(self) -> D;
+}
+
+impl<T, D> IntoIntoDart<Vec<D>> for Vec<T>
+where
+    T: IntoIntoDart<D>,
+{
+    fn into_into_dart(mut self) -> Vec<D> {
+        self.drain(0..).map(|e| e.into_into_dart()).collect()
+    }
+}
+
+impl<A, AD, B, BD> IntoIntoDart<(AD, BD)> for (A, B)
+where
+    A: IntoIntoDart<AD>,
+    B: IntoIntoDart<BD>,
+{
+    fn into_into_dart(self) -> (AD, BD) {
+        (self.0.into_into_dart(), self.1.into_into_dart())
+    }
+}
+
+impl<T, D> IntoIntoDart<Option<D>> for Option<T>
+where
+    T: IntoIntoDart<D>,
+{
+    fn into_into_dart(self) -> Option<D> {
+        self.map(|e| e.into_into_dart())
+    }
+}
+
+impl<T> IntoIntoDart<RustOpaque<T>> for RustOpaque<T>
+where
+    T: DartSafe,
+{
+    fn into_into_dart(self) -> RustOpaque<T> {
+        self
+    }
+}
+
+impl<T, D> IntoIntoDart<ZeroCopyBuffer<D>> for ZeroCopyBuffer<T>
+where
+    T: IntoIntoDart<D>,
+{
+    fn into_into_dart(self) -> ZeroCopyBuffer<D> {
+        ZeroCopyBuffer(self.0.into_into_dart())
+    }
+}
+
+impl<T, const C: usize> IntoIntoDart<[T; C]> for [T; C]
+where
+    T: IntoDart,
+{
+    fn into_into_dart(self) -> [T; C] {
+        self
+    }
+}
+
+impl<T> IntoIntoDart<T> for Box<T>
+where
+    T: IntoDart,
+{
+    fn into_into_dart(self) -> T {
+        *self
+    }
+}
+
+// more generic impls do not work because they crate possibly conflicting trait impls
+// this is why here are some more specific impls
+
+// Implementations for simple types
+macro_rules! impl_into_into_dart {
+    ($t:ty) => {
+        impl IntoIntoDart<$t> for $t {
+            fn into_into_dart(self) -> $t {
+                self
+            }
+        }
+    };
+}
+
+impl_into_into_dart!(u8);
```

Comment:
nit: btw did you copy it from somewhere? if so it is great to briefly point to there so it is easier to keep code in sync

Reply:
not really copied but i looked at the impls for `IntoDart`. I will add a comment for that